pkg/plugins/diff/delta_diff_server/src/delta_ops.rs

```diff
@@ -0,0 +1,75 @@
+use std::collections::HashMap;
+
+use deltalake::{DeltaTable, DeltaTableBuilder, DeltaTableError};
+use serde_json::{Map, Value};
+use tonic::{Code, Status};
+
+use crate::differ::OperationType;
+use crate::differ::TablePath;
+
+pub(crate) async fn get_delta_table(config: &HashMap<String, String>, repo: &str, table_path: TablePath) -> Result<DeltaTable, Status> {
+    let path = create_s3_path(repo, table_path);
+    return match create_table_with_config(config, path).await {
+        Ok(table) => {
+            Ok(table)
+        },
+        Err(err) => {
+            return Err(Status::new(Code::NotFound, format!("{:?}", err)));
+        }
+    };
+}
+
+fn create_s3_path(repo: &str, table_path: TablePath) -> String {
+    format!("s3://{}/{}/{}", repo, table_path.r#ref, table_path.path)
+}
+
+async fn create_table_with_config(config: &HashMap<String, String>, path: String) -> Result<DeltaTable, DeltaTableError> {
+    let cloned_config = config.clone();
+    let builder = DeltaTableBuilder::from_uri(path)
+        .with_storage_options(cloned_config);
+
+    builder.load().await
+}
+
+pub(crate) async fn history(delta: &mut DeltaTable, limit: Option<usize>) -> Result<Vec<Map<String, Value>>, Status> {
```

Comment:
Why is "delta" mut? Also, I see other muts in the code, which could be redundant.